### PR TITLE
refactor: removed unnecessary todo comment, modified jsdoc of utility func

### PR DESCRIPTION
**Pull request by tanmay4u02**
Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

A todo task and fixed two typescript errors in src/utils.ts
Moved ```statusToString``` function to utils since repeated at two places

Also needed to do 2 other changes in ```utils.ts``` to fix 2 typescript errors. The errors are pasted below in screenshots

This is the todo comment
https://github.com/louislam/uptime-kuma/blob/e773e454e9bfcf331095c9f88312bbf7b2e90359/server/notification-providers/dingding.js#L90
Did as mentioned here but moving it to ```NotificationProvider``` class would have resulted in in it being included in a lot of child classes extending ```NotificationProvider``` so moved it to utils

Fixes #(issue)

## Type of change

Please delete any options that are not relevant.

- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Screenshot 2024-10-07 122209

Screenshot 2024-10-07 122435

@louislam

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -331,7 +348,7 @@ class Logger {
      * @param msg Message to write
      * @returns {void}
      */
-    debug(module: string, msg: string): void {
+    debug(module: string, msg: unknown): void {
```
Comment (CommanderStorm) [inline src/util.ts:351]: Please change this upsteam instead.
The logging function should not need to handle non-strings.

File: src/util.ts [hunk 2]
```diff
@@ -120,7 +115,11 @@ export const badgeConstants = {
     defaultCertExpireDownDays: "7"
 };
 
-/** Flip the status of s */
+/**
+ * Flip the status of s
+ * @param s input status: UP or DOWN
+ * @returns {number} UP or DOWN
```
Comment (CommanderStorm) [inline src/util.ts:121]: Not quite correct.
```suggestion
 * Flip the status of s between UP and DOWN if this is possible
 * @param s {number} status
 * @returns {number} flipped status
```

File: server/notification-providers/dingding.js [hunk 3]
```diff
@@ -80,23 +80,6 @@ class DingDing extends NotificationProvider {
...
             .digest("base64");
     }
-
-    /**
-     * Convert status constant to string
-     * @param {const} status The status constant
-     * @returns {string} Status
-     */
-    statusToString(status) {
-        // TODO: Move to notification-provider.js to avoid repetition in classes
```
Comment (CommanderStorm) [inline server/notification-providers/dingding.js:90]: repeating ourselves twice out of 100 notification providers is not bad. Lets revert this change and remove the TODO.

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Hi, thanks for the PR.
I have left a few comments.

File: server/notification-providers/dingding.js (hunk 3 above)
Comment (tanmay4u02) [reply server/notification-providers/dingding.js:90]: Yes, that's why I have not moved it to Notification provider base class.

I have moved it to utils so we only have it once now. And it is more like a utility function rather than being a method of any specific class.

Sorry if I'm not able to interpret your comment correctly. Please correct me if I'm wrong

File: src/util.ts (hunk 1 above)
Comment (tanmay4u02) [reply src/util.ts:351]: Thank you for the comment. I changed its type to unknown because we're passing it directly to this.log function
image

where its type is given as unknown.
image

We are giving it a non-string msg to log
image

Here it was giving typescript error so I changed it to unknow.
Having string type was giving typescript error that I have added in screenshots

Please let me know if this is incorrect approach or if there is any other way

**Comment by tanmay4u02** [comment]
> * A lot of unexpected code changes.
> * I think we cannot test AliyunSMS / Dingding, may need Chinese users test it.

Thanks for the comment. Please let me know if the change is unnecessary. I'll revert the changes and remove the todo comment.

I had tested it by giving a random url and logging the moved function's response. It doesn't complete the process of sending notification due to incorrect url but the function calls are made

File: src/util.ts (hunk 2 above)
Comment (tanmay4u02) [reply src/util.ts]: Done this change. Thanks

File: src/util.ts (hunk 1 above)
Comment (CommanderStorm) [reply src/util.ts:349]: I meant that `this.hideLog` should be changed, instead of `debug`

File: server/notification-providers/dingding.js (hunk 3 above)
Comment (CommanderStorm) [reply server/notification-providers/dingding.js:90]: I meant that you should move this back to the individual notification providers where it came from and remove the TODO.

File: src/util.ts (hunk 1 above)
Comment (tanmay4u02) [reply src/util.ts:349]: Ok yes, changing ```this.hidelog``` with ```json.stringify``` of it would be correct.

But louislam has already done a change at this location so I'll revert this change

File: server/notification-providers/dingding.js (hunk 3 above)
Comment (tanmay4u02) [reply server/notification-providers/dingding.js:90]: Ok got it, reverted the changes and removed the TODO comment. Can you please check now

**Review by CommanderStorm — APPROVED** [review]
Seems much better, thanks for helping with the cleanup

**Comment by tanmay4u02** [comment]
> Seems much better, thanks for helping with the cleanup

Thank you
